### PR TITLE
GeoJSON datasets

### LINKED ISSUE #191: Add full precision GeoJSON version for TopoJSON only datastes
Elastic Maps will always prefer TopoJSON as a convenient format with a good balance between data size and precision. Still, GeoJSON has a broader adoption in the industry, so we will provide a better service to our users if we offer our files in that format additionally.

The proposal then is for our current TopoJSON datasets (World Countries, Admin Regions, USA counties) to also be exposed in GeoJSON format in the EMS  manifest. This format will be preferred by EMS Landing Page at <maps.elastic.co>.

This task is affected by https://github.com/elastic/kibana/issues/71975 and related to https://github.com/elastic/ems-landing-page/issues/255

### PR DESCRIPTION
**Pull request by jsanz**
Fixes #191 

This PR includes the following new GeoJSON data sources, with the same precision that their already existing TopoJSON versions.

* `admin_regions_lvl2_v2.geo.json` 25MB
* ` usa_counties_v2.geo.json` 2MB
* `usa_zip_codes_v7.geo.json` 11MB
* `world_countries_v7.geo.json` 11MB

Changes in existing admin regions and world countries are related to a small fix in the data pipeline to include proper IDs that I did as part of improving the `Makefile`. 

For the World Countries, the new datasets are only available from `7.10`, offering to the older versions the current published low precision dataset.

### CODE REVIEW

File: sources/world/Makefile [hunk 1]
```diff
@@ -1,12 +1,17 @@
...
+.DEFAULT_GOAL := all
+
+.PHONY: all
+all: ../../data/admin_regions_lvl2_v2.topo.json ../../data/admin_regions_lvl2_v2.geo.json ../../data/admin_regions_lvl2_v1.geo.json ../../data/world_countries_v7.topo.json ../../data/world_countries_v7.geo.json ../../data/world_countries_v1.geo.json # Main target for all resulting datasets
 
 .PHONY: clean
 clean: ## Remove output files
 	rm -f $(DATA_DIR)/admin_regions_lvl2_v1.geo.json \
   $(DATA_DIR)/admin_regions_lvl2_v2.topo.json \
+  $(DATA_DIR)/admin_regions_lvl2_v2.geo.json \
```
Comment (nickpeihl) [inline sources/world/Makefile:11]: Do we want to continue to use `$(DATA_DIR)` here? I know we switched back to using relative paths in other places.

**Review by nickpeihl — CHANGES_REQUESTED** [review]
Can you double-check that you have the latest shapefiles from the Google drive? I'm seeing some missing regions such as AZ-LAC, MU-PU, and others in the admin_regions dataset. 

There remains some inconsistencies when dissolving the regions into countries. For example, the overseas territories of France (such as French Guiana) are completely missing from world countries. Also in world countries, Taiwan has been merged into China. 

Compare your PR to what's currently in staging.
```
vimdiff <(curl --compressed https://storage.googleapis.com/elastic-bekitzur-emsfiles-vector-dev/files/world_countries_v1.geo.json | jq '.features[].properties.iso2' | sort | uniq) <(jq '.features[].properties.iso2' < data/world_countries_v1.geo.json | sort | uniq)
```

File: sources/world/Makefile (hunk 1 above)
Comment (jsanz) [reply sources/world/Makefile:11]: yeah, the problem is that it cannot be used in the target names or dependencies, so not a lot of chances to benefit from it. I'll remove it.

File: sources/world/Makefile [hunk 2]
```diff
@@ -38,16 +43,29 @@ clean: ## Remove output files
...
   -sort this.properties.region_iso_code \
+  -clean \
   -o format=topojson id-field=region_iso_code prettify $(DATA_DIR)/admin_regions_lvl2_v2.topo.json
 
+../../data/admin_regions_lvl2_v2.geo.json: ../../data/admin_regions_lvl2_v2.topo.json ## Regions GeoJSON full size
+	mapshaper -i $(DATA_DIR)/admin_regions_lvl2_v2.topo.json \
+  -sort this.properties.region_iso_code \
+  -clean \
+  -o rfc7946 id-field=region_iso_code format=geojson force /tmp/admin_regions_lvl2_v2.geo.json
+	node ../../scripts/clean-geom.js -v /tmp/admin_regions_lvl2_v2.geo.json
```
[the remark this answers is not in the document]
Comment (jsanz) [reply sources/world/Makefile:54]: Agree that is out of scope but I'm not sure if the `-clean` command is fixing all issues.

**Comment by jsanz** [comment]
> Can you double-check that you have the latest shapefiles from the Google drive? I'm seeing some missing regions such as AZ-LAC, MU-PU, and others in the admin_regions dataset.

Sorry about that, I will double check the data, thanks!!

**Comment by jsanz** [comment]
@nickpeihl I've marked again this for review, hopefully data issues are addressed. I also changed the source for the world countries to only offer TopoJSON for `>= 7.10`.

**Review by nickpeihl — COMMENTED** [review]
World countries look good to me. 

Administrative regions are missing a few regions in Azerbaijan and Slovenia, but that's my fault because I did not rsync my local changes to the Google Drive. Sorry about that.

```
vimdiff <(curl --compressed https://vector.maps.elastic.co/files/admin_regions_lvl2_v1.geo.json | jq '.features[].properties.region_iso_code' | sort | uniq) <(jq '.features[].properties.region_iso_code' < data/admin_regions_lvl2_v1.geo.json | sort | uniq)
```

Getting rid of the `~99-` regions is a good idea though. Thanks for doing that.

**Comment by jsanz** [comment]
@nickpeihl thanks for updating the datasets, I've tested again the regions and now we don't have any missing records compared with current production datasets and the new ones are all under control either as current missing data, or new data as per processing to generate all countries.

Also, there are no diffs with the current production world countries dataset

```
diff \
 <(curl -s --compressed https://vector.maps.elastic.co/files/world_countries_v1.geo.json | jq '.features[].properties.iso2' | sort | uniq) \
 <(jq '.features[].properties.iso2' < data/world_countries_v1.geo.json | sort | uniq)
```

**Review by nickpeihl — APPROVED** [review]
lgtm

Code review and compared datasets in this PR to production

**Pull request merged**